### PR TITLE
Switches

### PR DESCRIPTION
**Pull request by thibaudk**
I also moved the Process loops to the "Edit processes" page as mentioned in the previous PR

### CODE REVIEW

File: docs/reference-manual/common-practices/2-switches.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+Combining both the above options, an interval can be toggled on and off at will. This method, like the previous, can also be directly mapped to various devices featuring buttons, keys, hardware switches or any kind of discrete value streams. In this case we will use the first button of a game controller accessed through the [Joystick Device](/score-docs/docs/references/devices-types/joystick-device.html).
+
+![Basic switch](/score-docs/assets/images/reference-manual/common-practices/switch.gif "Basic switch")
+
+When the button is pressed, the first trigger is validated (as by default, it is set to "pulse") and will respond to any value received at the provided address. When the button is released, the end trigger is validated in the same way, before the transition returns to the initial `Sync`.
+
+## Consecutive toggles
+
+The section inside the loop can contain more than one interval. Expanding on the prior exemple, switching between an idol state and an interval, a similar structure could enclose two intervals. This switch also requires an extra trigger.
```
Comment (jcelerier) [inline docs/reference-manual/common-practices/2-switches.md:47]: idol => idle

File: docs/reference-manual/common-practices/2-switches.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+When the button is pressed, the first trigger is validated (as by default, it is set to "pulse") and will respond to any value received at the provided address. When the button is released, the end trigger is validated in the same way, before the transition returns to the initial `Sync`.
+
+## Consecutive toggles
+
+The section inside the loop can contain more than one interval. Expanding on the prior exemple, switching between an idol state and an interval, a similar structure could enclose two intervals. This switch also requires an extra trigger. 
+
+![Consecutive switch](/score-docs/assets/images/reference-manual/common-practices/consecutive.gif "Consecutive switch")
+
+With the default "Pulse" operators to validate each trigger, this structure will return to the idol state after each transition. To avoid this extra step and make sure that the first interval executes every time the button's value is true and that the second executes every time it is false, the expression for the trigger has to be modified. By setting both the start and end triggers to only respond when the value is true, they can be validated in a row, skipping the idol state after the transition.
```
Comment (jcelerier) [inline docs/reference-manual/common-practices/2-switches.md:51]: idol => idle

File: docs/reference-manual/common-practices/2-switches.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+![Consecutive 2 stages switch](/score-docs/assets/images/reference-manual/common-practices/seq2Stage.gif "Consecutive 2 steps switch")
+
+A better way of achieving the same result would be to rely on a [sub-scenario to isolate the loop](/score-docs/docs/reference-manual/common-practices/1-looping.html#nested-loops). Since the first trigger is only needed to control the very first execution of the loop, it can be validated once to enter the sub-scenario and omitted in the actual toggle structure. The trigger at the end of the sub-scenario is left "never" valid to allow as many repetitions as needed.
+
+![Consecutive switch isolated](/score-docs/assets/images/reference-manual/common-practices/seqIsolate.gif "Consecutive switch isolated")
+
+In context it may be preferable to switch at evry press of the button, instead of having to hold for the first interval and let go for the second. Adding a minimum duration on each of the toggled intervals ensures that the value of the button can go back to false without validating the next trigger. This method also protects the switch from accidental "double taps".
```
Comment (jcelerier) [inline docs/reference-manual/common-practices/2-switches.md:59]: in context : ? 
evry => every

File: docs/reference-manual/common-practices/2-switches.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+![Tap once to switch](/score-docs/assets/images/reference-manual/common-practices/tapToSwitch.gif "Tap once to switch")
+
+## Parallel switches
+
+Consecutive switches shawn above can sequence an arbitrary number of intervals, but only in the same order every time. To enable switching as well as reordering, we can rely on conditional branches. Let's first create another device for control. For this example we will add a [Mapper Device](/score-docs/docs/references/devices-types/mapper-device.html) by simply copying and pasting the script below. We'll name it "Structure".
```
Comment (jcelerier) [inline docs/reference-manual/common-practices/2-switches.md:65]: shawn => shown

File: docs/reference-manual/common-practices/2-switches.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+![Parallel sync](/score-docs/assets/images/reference-manual/common-practices/parallelSync.gif "Parallel sync")
+
+To add conditions for each start state of parallel intervals, they have to be splitted into individual events with the "Split condition" fonction. The state connecting the transition back to the initial `Sync` does not require a condition. Finally, a trigger on the end `Sync` can reset the loop with every new value set in the mapper.
+
+![Conditions](/score-docs/assets/images/reference-manual/common-practices/conditions.gif "Conditions")
+
+It is now possible to switch between parallel intervals when setting a value to the "Branch" parameter. To control the start of the structure when played the first time, we have to avoid adding a trigger to the start sync of the loop. It would add an idol state as we experienced before and skip the evaluation of every other value. Instead we can isolate this switch from it's initial trigger with another transition, similar to our solution to [looping parallel branches](/score-docs/docs/reference-manual/common-practices/1-looping.html#loop-branches).
```
Comment (jcelerier) [inline docs/reference-manual/common-practices/2-switches.md:93]: idol => idle
 it's initial trigger => its initial trigger

File: docs/reference-manual/processes/2-edit.md [hunk 6]
```diff
@@ -27,4 +27,11 @@ To display the process in full size, double-click on the interval the process is
...
 To exit full size mode (go back to parent scenario), click the desired node of the path displayed at the top or use the {% include shortcut.html content="Ctrl+Alt+↑" %} shortcut.
 
-![Edit process in full size](/score-docs/assets/images/reference-manual/processes/process-fullsize-edit.gif "Edit process in full size")
\ No newline at end of file
+![Edit process in full size](/score-docs/assets/images/reference-manual/processes/process-fullsize-edit.gif "Edit process in full size")
+
+## Process Loops
+
+Time-based processes such as automations, media and sub-scenarios can be looped for as long as their parent interval is active. When selecting a process, either from the object panel 
+or by clicking on process header, the loop option will appear in the `Inspector`
```
Comment (jcelerier) [inline docs/reference-manual/processes/2-edit.md:35]: add a period at the end of the sentence

**Review by jcelerier — COMMENTED** [review]
thanks for the effort on the writing it's much better than the previous one !

File: docs/reference-manual/processes/2-edit.md (hunk 6 above)
Comment (thibaudk) [reply docs/reference-manual/processes/2-edit.md]: done with 4219439

File: docs/reference-manual/common-practices/2-switches.md (hunk 5 above)
Comment (thibaudk) [reply docs/reference-manual/common-practices/2-switches.md]: done with 4219439

File: docs/reference-manual/common-practices/2-switches.md (hunk 4 above)
Comment (thibaudk) [reply docs/reference-manual/common-practices/2-switches.md]: done with 4219439

My mom's nick name is Shawn ...

File: docs/reference-manual/common-practices/2-switches.md (hunk 3 above)
Comment (thibaudk) [reply docs/reference-manual/common-practices/2-switches.md]: done with 4219439

removed "In context", it's not needed

File: docs/reference-manual/common-practices/2-switches.md (hunk 2 above)
Comment (thibaudk) [reply docs/reference-manual/common-practices/2-switches.md]: done with 4219439

File: docs/reference-manual/common-practices/2-switches.md (hunk 1 above)
Comment (thibaudk) [reply docs/reference-manual/common-practices/2-switches.md]: done with 4219439

**Pull request merged**